service/src/main/java/bio/terra/workspace/common/logging/WorkspaceActivityLogHook.java

```diff
@@ -60,11 +69,20 @@ public HookAction endFlight(FlightContext context) {
       return HookAction.CONTINUE;
     }
 
+    var userRequest =
+        checkNotNull(
+            context
+                .getInputParameters()
+                .get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class));
+    var userStatusInfo = samService.getUserStatusInfo(userRequest);
```

Comment:
Have you already made this call once at a higher level before calling the flight?
Should we be serializing the user status info into the input parameters instead of asking Sam again?

Reply:
No and sometimes yes. only if it's a direct write into the database, I check sam for user status info. But sometimes it's in the same method depending on whether it needs a flight or is a direct write to db. 

do you want to serialize the user status info and only if it doesn't exist ask Sam?

Reply:
> do you want to serialize the user status info and only if it doesn't exist ask Sam?

I think we should. You could do it as a separate ticket. Sam is very heavily used, so we do a community service by avoiding extra calls.

Reply:
> can you use WorkspaceActivityLogService instead of calling SAM yourself?

SamService is a service wrapping and handling calls to Sam, so technically this is not calling Sam directly. We typically call our SamService in cases like this.